### PR TITLE
feat: improve debugging experience of workflow runs

### PR DESCRIPTION
**Pull request by abelanger5**
Makes steps in a workflow run expandable to view input, output and error information.

Also adds the triggering cron or event to the top of the workflow run.

Some screenshots:

Cron workflow (from examples directory):

image

Timeout example:

image

### CODE REVIEW

File: frontend/app/src/pages/main/workflow-runs/$run/index.tsx [hunk 1]
```diff
@@ -136,3 +164,204 @@ export default function ExpandedWorkflowRun() {
...
+}) {
+  return (
+    <TableRow key={stepRun.metadata.id}>
+      <TableCell colSpan={columns.length} className="px-8 py-4">
+        <StepStatusSection stepRun={stepRun} />
+        <StepConfigurationSection stepRun={stepRun} />
+        <StepInputSection stepRun={stepRun} />
+        <StepOutputSection stepRun={stepRun} />
+      </TableCell>
+      {/* <TableCell colSpan={1}>
```
Comment (abelanger5) [inline frontend/app/src/pages/main/workflow-runs/$run/index.tsx]: Remove unused

File: frontend/app/src/pages/main/workflow-runs/components/run-statuses.tsx [hunk 2]
```diff
@@ -17,6 +23,11 @@ export function RunStatus({ status }: { status: RunStatus }) {
     case "CANCELLED":
       variant = "failed";
       text = "Failed";
+
+      if (reason === "TIMED_OUT") {
```
Comment (steebchen) [inline frontend/app/src/pages/main/workflow-runs/components/run-statuses.tsx]: nit: maybe `switch` would be better for different future reasons

File: frontend/app/src/pages/main/workflow-runs/$run/index.tsx [hunk 3]
```diff
@@ -136,3 +164,193 @@ export default function ExpandedWorkflowRun() {
...
+    <>
+      <Code language="json" className="my-4" maxHeight="400px">
+        {JSON.stringify(JSON.parse(eventData.data), null, 2)}
+      </Code>
+    </>
+  );
+}
+
+function TriggeringCronSection({ cron }: { cron: string }) {
+  // const interval = CronParser.parseExpression(cron, {
```
Comment (steebchen) [inline frontend/app/src/pages/main/workflow-runs/$run/index.tsx]: nit: remove?

**Review by steebchen — APPROVED** [review]
LGTM

File: frontend/app/src/pages/main/workflow-runs/components/run-statuses.tsx (hunk 2 above)
Comment (abelanger5) [reply frontend/app/src/pages/main/workflow-runs/components/run-statuses.tsx]: Fixed this and added some additional cases for when a previous step run timed out.

**Pull request merged**